CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportDialogManager.java

```diff
@@ -0,0 +1,197 @@
+    private final ArrayList<String> popupTypeFilters = new ArrayList<>();
+    private boolean isErrorReportRunning = false;
+    private Timer refreshTimer = null;
+    private Date latestRetrievalDate = null;
+    private Date previousRetrievalDate = null;
+    private Date backupRetrievalDate = null;
+    private Date gracePeriodResumptionDate = null;
+    private ArrayList<ErrorReportEntry> sessionPopupErrors = new ArrayList<>();
+
+    private static ErrorReportDialogManager instance = null;
+    
+    public ErrorReportDialogManager(){
+    }
+
+    public static ErrorReportDialogManager getInstance() {
+        if (instance == null) {
+            instance = new ErrorReportDialogManager();
+            instance.startDefaultProcessor();
+        }
+        return instance;
+    }
+
+    public void startDefaultProcessor(){
+        popupTypeFilters.add("SEVERE");
+        popupTypeFilters.add("WARNING");
+        final TimerTask refreshAction = new TimerTask() {
+            @Override
+            public void run() {
+                if (!isErrorReportRunning) {
+                    Platform.runLater(new Runnable() {
+                        public void run() {
+                            final Date currentDate = new Date();
+                            if (gracePeriodResumptionDate != null && currentDate.before(gracePeriodResumptionDate)) {
+                                return;
+                            }
+                            if (latestRetrievalDate == null || latestRetrievalDate.before(ErrorReportSessionData.lastUpdate)) {
+                                backupRetrievalDate = previousRetrievalDate == null ? null : new Date(previousRetrievalDate.getTime());
+                                previousRetrievalDate = latestRetrievalDate == null ? null : new Date(latestRetrievalDate.getTime());
+                                latestRetrievalDate = currentDate;
+                                sessionPopupErrors = ErrorReportSessionData.getInstance().refreshDisplayedErrors(popupTypeFilters);
+                                for (final ErrorReportEntry reportEntry : sessionPopupErrors) {
+                                    if (popupTypeFilters.contains(reportEntry.getErrorLevel().getName())) {
+                                        if (previousRetrievalDate == null || reportEntry.getLastPopupDate() == null || reportEntry.getLastDate().after(previousRetrievalDate)) {
+                                            showErrorDialog(reportEntry);
+                                        }
+                                    }
+                                }
+
+                            }
+                        }
+                    });
+                }
+            }
+        };
+        refreshTimer = new Timer();
+        refreshTimer.schedule(refreshAction, 175, 475);        
+    }
+    
+    public void updatePopupSettings(final int popupMode, ArrayList<String> popupFilters) {
```

Comment:
`popupFilters` can be final. Also if you could make it just a List rather than ArrayList